src/containers/FullApp.jsx

```diff
@@ -148,13 +157,56 @@ export class FullApp extends Component {
     }
 
     loadPatient(patientId) {
-        let patient = this.dataAccess.getPatient(patientId);
-        this.contextManager = new ContextManager(patient, this.onContextUpdate);
-        this.setState({patient: patient})
+        if (this.dataAccess.getGestalt().requestTypes.async) { 
+            this.dataAccess.getPatient(patientId, (patient, error) => { 
+                this.contextManager = new ContextManager(patient, this.onContextUpdate);
+                // TODO: REMOVE AFTER TESTING
+                setTimeout(()=>this.setState({ 
+                    patient, 
+                    loading: false,
+                    loadingErrorObject: error
+                }), 5000);
+                //
+                // TODO: RESTORE AFTER TESTING
+                // this.setState({ 
+                //     patient, 
+                //     loading: false,
+                //     loadingErrorObject: error
+                // });
+            });
+        } else { 
+            // Else, assume sync
+            try {
+                let patient = this.dataAccess.getPatient(patientId);
+                this.contextManager = new ContextManager(patient, this.onContextUpdate);
+                // TODO: REMOVE AFTER TESTING
+                setTimeout(()=>this.setState({ 
+                    patient, 
+                    loading: false,
+                }), 5000);
+                //
+                // TODO: RESTORE AFTER TESTING
+                // this.setState({
+                //     patient, 
+                //     loading: false
+                // });
+            } catch (error) {
```

Comment:
If there is an error somewhere, nothing from here on gives you easy access to the full stack trace, unless you're already debugging. Same with the async path above. The LoadingError object just shows the message which isn't always sufficient.
I'm not sure if there's a Flux standard for error logging, but it could be helpful to write the error to console.error either here or in the LoadingError class, that way all the individual DataSource classes don't necessarily have to wrap everything in try/catch with a logger. Compare to before when there was no error handler and any errors would just bubble up.

Reply:
I like it -- implementing now.

Reply:
Nice. I added that functionality into the different apps, so that's logged properly as errors.